searches/binary_search.py

```diff
@@ -297,6 +297,49 @@ def binary_search_by_recursion(
         return binary_search_by_recursion(sorted_collection, item, midpoint + 1, right)
 
 
+def exponential_search(sorted_collection: list[int], item: int) -> int | None:
```

Comment:
I am not a fan of functions that return multiple datatypes.  If the search fails, let's return `-1` (like `"Hello".find("Z")`) or raise a `ValueError` (like `"Hello".index("Z")`).
```suggestion
def exponential_search(sorted_collection: list[int], item: int) -> int:
```